src/pipx/commands/ensure_path.py

```diff
@@ -1,42 +1,140 @@
 import logging
 from pathlib import Path
+import site
+import sys
+import textwrap
+from typing import Tuple, Optional
 
 import userpath  # type: ignore
 from pipx.emojies import stars
+from pipx import constants
 
 
-def ensure_path(location: Path, *, force: bool):
+def wrap_indent(text: str) -> str:
+    text_wrapped = textwrap.wrap(text)
+    if len(text_wrapped) > 1:
+        return (
+            text_wrapped[0]
+            + "\n"
+            + textwrap.indent("\n".join(text_wrapped[1:]), "    ")
+        )
+    else:
+        return text_wrapped[0]
+
+
+def get_pipx_user_bin_path() -> Optional[Path]:
+    """Returns None if pipx is not installed using `pip --user`
+    Otherwise returns parent dir of pipx binary
+    """
+    # NOTE: using this method to detect pip user-installed pipx will return
+    #   None if pipx was installed as editable using `pip install --user -e`
+
+    # https://docs.python.org/3/install/index.html#inst-alt-install-user
+    #   Linux + Mac:
+    #       scripts in <userbase>/bin
+    #   Windows:
+    #       scripts in <userbase>/Python<XY>/Scripts
+    #       modules in <userbase>/Python<XY>/site-packages
+
+    pipx_bin_path = None
+
+    script_path = Path(__file__).resolve()
+    userbase_path = Path(site.getuserbase()).resolve()
+    try:
+        _ = script_path.relative_to(userbase_path)
+    except ValueError:
+        pip_user_installed = False
+    else:
+        pip_user_installed = True
+    if pip_user_installed:
+        test_paths = (
+            userbase_path / "bin" / "pipx",
+            Path(site.getusersitepackages()).resolve().parent / "Scripts" / "pipx.exe",
+        )
+        for test_path in test_paths:
+            if test_path.exists():
+                pipx_bin_path = test_path.parent
+                break
+
+    return pipx_bin_path
```

Comment:
```python
import os
import sysconfig

if os.name == "posix":
    scheme = "posix_user"
elif os.name == "nt":
    scheme = "nt_user"
else:
    raise RuntimeError(f"unsupported operating system {os.name!r}")

dirpath = Path(sysconfig.get_path("scripts", scheme=scheme))
for name in ["pipx", "pipx.exe"]:
    if Path(dirpath, name).exists():
        return dirpath
```

Reply:
This seems good to me also.  I can modify the code to use sysconfig if people would like me to.

Reply:
Is this resolved?

Reply:
I wasn't sure if @uranusjr was just building awareness for `sysconfig` or if he wanted me to change it. 🙂  I was going to leave it, but can change it if people wish.

Reply:
Either way is fine with me